sorting/selection_sort_recursive.c

```diff
@@ -0,0 +1,109 @@
+/**
+ * @file
+ * @author [Dhruv Pasricha](https://github.com/DhruvPasricha)
+ * @brief [Selection Sort](https://en.wikipedia.org/wiki/Selection_sort)
+ * implementation using recursion.
+ */
+ 
+#include <assert.h>     /// for assert
+#include <stdio.h>      /// for IO operations
+#include <stdlib.h>     /// for dynamic memory allocation
+#include <time.h>       /// for random numbers generation
+#include <inttypes.h>   /// for uint8_t
+
+/**
+ * @brief Swapped two numbers using pointer
+ * @param first pointer of first number
+ * @param second pointer of second number
+ */
+void swap(uint8_t *first, uint8_t *second)
```

Comment:
Selection sort can also sort negative numbers, therefore please use singed integers in arguments.